docs/design/hts_address_tx_receipt.md

```diff
@@ -0,0 +1,62 @@
+# HTS Token Creation via SystemContract Receipt Enhancement
+
+## Purpose
+The purpose of this enhancement is to improve the interoperability between Hedera Token Service (HTS) and Ethereum-compatible tools, like ethers, by ensuring access to the necessary information about newly created tokens, in this case token address. This change specifically addresses the gap in Web3 tooling support when creating tokens through HTS precompile contracts, making user experience smoother.
+
+
+## Problem Statement
+When users create tokens (fungible and non-fungible) by directly calling the 0x167 contract there is no way for them to get the evm address of the newly created token via standard EVM tooling e.g ethers. This is because the transaction receipt's contractAddress field is not being populated with the HTS token address, since essentially the operations is just a contract call, but not a contract create (deploy).
+
+## Current Behaviour
+
+Currently, when creating a token via HTS the address of the system contract is returned as a contractAddress in the transaction receipt.
+
+## Proposed Solution
+
+### Technical Details
+
+### 1. Function Signature Detection
+
+Detect in the transactionResponse from the mirror node, if the transaction was calling any of the HTS methods - createFungibleToken/createNonFungibleToken/createFungibleTokenWithCustomFees/createNonFungibleTokenWithCustomFees
+
+N.B Currently, HTS supports both v1 and v2 security model function selectors
+
+```javascript
+const HTS_CREATE_FUNCTIONS_SIGNATURE = [
+  "createFungibleToken(...)",
+  "createNonFungibleToken(...)",
+  "createFungibleTokenWithCustomFees(...)",
+  "createNonFungibleTokenWithCustomFees(...)"
+];
```

Comment:
If possible, let's include the correct params for these methods instead of `(...)`